File: vaiz/api/boards.py
from vaiz.api.base import BaseAPIClient
from vaiz.models import (
    BoardsResponse, 
    BoardResponse, 
    CreateBoardTypeRequest, 
    CreateBoardTypeResponse,
    EditBoardTypeRequest,
    EditBoardTypeResponse,
    CreateBoardCustomFieldRequest,
    CreateBoardCustomFieldResponse,
    EditBoardCustomFieldRequest,
    EditBoardCustomFieldResponse,
    CreateBoardGroupRequest,
    CreateBoardGroupResponse,
    EditBoardGroupRequest,
    EditBoardGroupResponse,
)


class BoardsAPIClient(BaseAPIClient):
    def get_boards(self) -> BoardsResponse:
        """
        Get all boards in the current space.
        
        Returns:
            BoardsResponse: The list of boards
        """
        response_data = self._make_request("getBoards", method="POST", json_data={})
        return BoardsResponse(**response_data)

    def get_board(self, board_id: str) -> BoardResponse:
        """
        Get a single board by its ID.
        
        Args:
            board_id (str): The ID of the board to retrieve
            
        Returns:
            BoardResponse: The board information
        """
        response_data = self._make_request("getBoard", method="POST", json_data={"boardId": board_id})
        return BoardResponse(**response_data)

    def create_board_type(self, request: CreateBoardTypeRequest) -> CreateBoardTypeResponse:
        """
        Create a new board type.
        
        Args:
            request (CreateBoardTypeRequest): The board type creation request
            
        Returns:
            CreateBoardTypeResponse: The created board type information
        """
        response_data = self._make_request("createBoardType", method="POST", json_data=request.model_dump())
        return CreateBoardTypeResponse(**response_data)

    def edit_board_type(self, request: EditBoardTypeRequest) -> EditBoardTypeResponse:
        """
        Edit an existing board type.
        
        Args:
            request (EditBoardTypeRequest): The board type edit request
            
        Returns:
            EditBoardTypeResponse: The updated board type information
        """
        response_data = self._make_request("editBoardType", method="POST", json_data=request.model_dump())
        return EditBoardTypeResponse(**response_data)

    def create_board_custom_field(self, request: CreateBoardCustomFieldRequest) -> CreateBoardCustomFieldResponse:
        """
        Create a new custom field in a board.
        
        Args:
            request (CreateBoardCustomFieldRequest): The custom field creation request
            
        Returns:
            CreateBoardCustomFieldResponse: The created custom field information
        """
        response_data = self._make_request("createBoardCustomField", method="POST", json_data=request.model_dump())
        return CreateBoardCustomFieldResponse(**response_data)

    def edit_board_custom_field(self, request: EditBoardCustomFieldRequest) -> EditBoardCustomFieldResponse:
        """
        Edit an existing custom field in a board.
        
        Args:
            request (EditBoardCustomFieldRequest): The custom field edit request
            
        Returns:
            EditBoardCustomFieldResponse: The updated custom field information
        """
        response_data = self._make_request("editBoardCustomField", method="POST", json_data=request.model_dump())
        return EditBoardCustomFieldResponse(**response_data)

    def create_board_group(self, request: CreateBoardGroupRequest) -> CreateBoardGroupResponse:
        """
        Create a new group in a board.

        Args:
            request (CreateBoardGroupRequest): The board group creation request

        Returns:
            CreateBoardGroupResponse: The list of board groups including the new one.
        """
        response_data = self._make_request("createBoardGroup", method="POST", json_data=request.model_dump())
        return CreateBoardGroupResponse(**response_data)

    def edit_board_group(self, request: EditBoardGroupRequest) -> EditBoardGroupResponse:
        """
        Edit a group in a board.

        Args:
            request (EditBoardGroupRequest): The board group edit request

        Returns:
            EditBoardGroupResponse: The list of board groups after editing.
        """
        response_data = self._make_request("editBoardGroup", method="POST", json_data=request.model_dump())
        return EditBoardGroupResponse(**response_data) 

File: vaiz-ts-sdk/src/boards.ts
import { BaseAPIClient } from './base';
import {
  BoardsResponse,
  BoardResponse,
  CreateBoardTypeRequest,
  CreateBoardTypeResponse,
  EditBoardTypeRequest,
  EditBoardTypeResponse,
  CreateBoardCustomFieldRequest,
  CreateBoardCustomFieldResponse,
  EditBoardCustomFieldRequest,
  EditBoardCustomFieldResponse,
  CreateBoardGroupRequest,
  CreateBoardGroupResponse,
  EditBoardGroupRequest,
  EditBoardGroupResponse,
} from './types';

export class BoardsAPIClient extends BaseAPIClient {
  
  async getBoards(): Promise<BoardsResponse> {
    /**
     * Get all boards in the current space.
     * 
     * @returns The list of boards
     */
    const responseData = await this.request<BoardsResponse>('getBoards', 'POST', {});
    return responseData;
  }

  async getBoard(boardId: string): Promise<BoardResponse> {
    /**
     * Get a single board by its ID.
     * 
     * @param boardId The ID of the board to retrieve
     * @returns The board information
     */
    const responseData = await this.request<BoardResponse>(
      'getBoard',
      'POST',
      { boardId }
    );
    return responseData;
  }

  async createBoardType(request: CreateBoardTypeRequest): Promise<CreateBoardTypeResponse> {
    /**
     * Create a new board type.
     * 
     * @param request The board type creation request
     * @returns The created board type information
     */
    const responseData = await this.request<CreateBoardTypeResponse>(
      'createBoardType',
      'POST',
      request
    );
    return responseData;
  }

  async editBoardType(request: EditBoardTypeRequest): Promise<EditBoardTypeResponse> {
    /**
     * Edit an existing board type.
     * 
     * @param request The board type edit request
     * @returns The updated board type information
     */
    const responseData = await this.request<EditBoardTypeResponse>(
      'editBoardType',
      'POST',
      request
    );
    return responseData;
  }

  async createBoardCustomField(request: CreateBoardCustomFieldRequest): Promise<CreateBoardCustomFieldResponse> {
    /**
     * Create a new custom field in a board.
     * 
     * @param request The custom field creation request
     * @returns The created custom field information
     */
    const responseData = await this.request<CreateBoardCustomFieldResponse>(
      'createBoardCustomField',
      'POST',
      request
    );
    return responseData;
  }

  async editBoardCustomField(request: EditBoardCustomFieldRequest): Promise<EditBoardCustomFieldResponse> {
    /**
     * Edit an existing custom field in a board.
     * 
     * @param request The custom field edit request
     * @returns The updated custom field information
     */
    const responseData = await this.request<EditBoardCustomFieldResponse>(
      'editBoardCustomField',
      'POST',
      request
    );
    return responseData;
  }

  async createBoardGroup(request: CreateBoardGroupRequest): Promise<CreateBoardGroupResponse> {
    /**
     * Create a new group in a board.
     * 
     * @param request The group creation request
     * @returns The created group information
     */
    const responseData = await this.request<CreateBoardGroupResponse>(
      'createBoardGroup',
      'POST',
      request
    );
    return responseData;
  }

  async editBoardGroup(request: EditBoardGroupRequest): Promise<EditBoardGroupResponse> {
    /**
     * Edit an existing group in a board.
     * 
     * @param request The group edit request
     * @returns The updated group information
     */
    const responseData = await this.request<EditBoardGroupResponse>(
      'editBoardGroup',
      'POST',
      request
    );
    return responseData;
  }
}